### PR TITLE
mgr/dashboard: remove duplicate ids from table column filters

### PR DESCRIPTION
**Pull request by nSedrickm**
Generate random ids for column filters to avoid id conflics on pages with multiple tables sharing the same column filters

Signed-off-by: nsedrickm <nsedrick101@gmail.com>
Fixes: https://tracker.ceph.com/issues/55869

Before
image

image

After
image

image

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [x] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

**Comment by nSedrickm** [comment]
jenkins test make check

File: src/pybind/mgr/dashboard/frontend/src/app/shared/datatable/table/table.component.html [hunk 1]
```diff
@@ -114,10 +114,10 @@
                 <input class="custom-control-input"
                        type="checkbox"
                        [name]="column.prop"
-                       [id]="column.prop"
+                       [id]="column.prop + tableName"
```
Comment (epuertat) [inline src/pybind/mgr/dashboard/frontend/src/app/shared/datatable/table/table.component.html:117]:
```suggestion
                       id="{{ column.prop }}{{ tableName }}"
```

**Comment by nSedrickm** [comment]
jenkins test dashboard cephadm

**Review by ljflores — APPROVED** [review]

**Comment by nSedrickm** [comment]
jenkins test dashboard cephadm

**Comment by nSedrickm** [comment]
jenkins test dashboard cephadm

**Comment by nSedrickm** [comment]
Hi @pereman2, all good here with one more review!

**Review by pereman2 — APPROVED** [review]

**Pull request merged**